src/k2/httpproxy/HTTPProxy.cpp

```diff
@@ -456,6 +466,16 @@ seastar::future<> HTTPProxy::gracefulStop() {
 seastar::future<> HTTPProxy::start() {
     _registerMetrics();
     _registerAPI();
+    // Run timer at interval minimum of 1s and txn_timeout/2.
+    // After expiry it may take additional min(1s,timeout/2) for cleanup.
+    auto interval = _txnTimeout()/2;
+    if (interval > 1s) interval = 1s;
```

Comment:
comment says min 1s, but here you're capping it at 1s (i.e. the max). Did you mean to compare < ? Also, this check is probably not needed - we're configuring this on the cmdline anyway. You can log an Info message to say what interval is used

Reply:
Changed to use config value instead of trying to calculate interval based on txn timeout.